skills/wordpress-code-reference/search.py

```
@@ -0,0 +1,107 @@
+    url = f"{API_BASE_URL}/search?{urllib.parse.urlencode(params)}"
+
+    try:
+        with urllib.request.urlopen(url, timeout=REQUEST_TIMEOUT) as response:
+            data = json.loads(response.read().decode())
+    except urllib.error.HTTPError as e:
+        raise RuntimeError(f"HTTP error {e.code}") from e
+    except urllib.error.URLError as e:
+        raise RuntimeError(f"Network error: {e.reason}") from e
+
+    if not isinstance(data, list):
+        raise RuntimeError(data.get("message", "Unexpected API response"))
```

Ruff (0.14.14)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)